src/server/db/user.ts

```diff
@@ -26,13 +27,12 @@ export const insertUserWithAddress = async (address: string) => {
 };
 
 export const selectMatchPlayedByUser = async (userId: string) => {
-  const selectedUsers = await db
+  const matchesPlayed = await db
     .select()
-    .from(matches)
-    .where(
-      sql`${matches.id} IN (SELECT unnest(${users.matchesPlayed}) AS id FROM ${users} WHERE id = ${userId})`,
-    );
-  return selectedUsers;
+    .from(usersToMatches)
```

Comment:
If I recall correctly, the correct way to do such a join would be to select from users table, then inner join with the usersToMatches then a second inner join with matches